pipeline/test_transform.py

```diff
@@ -0,0 +1,152 @@
+"""Tests for transform functions."""
+
+from transform import convert_unix_to_datetime, clean_data, transform_sales_data
+import pytest
+
+
+def test_convert_unix_to_datetime_valid():
+    """Tests that a valid Unix timestamp is converted to a string.."""
+
+    unix_timestamp = 1718801551.04217
+
+    assert convert_unix_to_datetime(unix_timestamp) == "2024-06-19 12:52:31"
+
+
+def test_convert_unix_to_datetime_invalid_type():
+    """Tests that a TypeError is raised if a float is not passed in."""
+
+    unix_timestamp = "171880151.04217"
+
+    with pytest.raises(TypeError) as exc_info:
+        convert_unix_to_datetime(unix_timestamp)
+        assert (
+            str(exc_info.value)
+            == "Expected a float for unix_timestamp, but got <class 'str'>"
+        )
+
+
+def test_convert_unix_to_datetime_invalid_unix():
+    """Tests that a ValueError is raised when a invalid Unix is passed in."""
+
+    unix_timestamp = -1718801551.04217
+
+    with pytest.raises(ValueError) as exc_info:
+        convert_unix_to_datetime(unix_timestamp)
+    assert str(exc_info.value) == "Unix timestamp cannot be negative."
+
+
+def test_convert_unix_to_datetime_invalid_datetime():
+    """Tests that a ValueError is raised if an 'impossible' unix is passed in."""
+
+    unix_timestamp = 9918801551.04217
+
+    with pytest.raises(ValueError) as exc_info:
+        convert_unix_to_datetime(unix_timestamp)
+    assert str(exc_info.value).startswith(
+        "The provided Unix timestamp corresponds to a future date or time: "
+    )
+
+
+def test_clean_data_valid():
+    """Tests that the correct keys are removed from the dictionary."""
+
+    input_data = {
+        "slug_type": "track",
+        "track_album_slug_text": "some_slug",
+        "artist": "Artist Name",
+        "track_title": "Song Title",
+        "currency": "USD",
+        "amount_paid": 1.99,
+        "country_code": "US",
+        "item_price": 1.29,
+        "art_id": 12345,
+        "releases": [{"id": 67890}],
+        "package_image_id": 98765,
+        "amount_paid_fmt": "$1.99",
+        "art_url": "https://example.com/image.jpg",
+        "amount_over_fmt": "$0.70",
+        "item_slug": "song-title",
+    }
+
```

Comment:
Could make a class so that you don't have to copy and paste input_data over and over

Reply:
Good idea, but I think it wouldn't be necessary since this particular input data is only used once.

Reply:
Good idea, added in [d8a3aa9](https://github.com/Zhi-704/c11-apollo-bandcamp-tracker/pull/33/commits/d8a3aa9d48ffa9ae0af83619f43ccb598847620b)